src/test/java/teammates/logic/core/FeedbackResponseCommentsLogicTest.java

```diff
@@ -346,6 +353,365 @@ public void testGetFeedbackResponseCommentsForQuestionInSection_withSectionName_
+
+        // comment visibility follows feedback question's visibility
+        FeedbackResponseCommentAttributes commentFromS4C1ToR2Q2S1C1VisibilityFollowsQuestion =
+                FeedbackResponseCommentAttributes.builder()
+                .withCommentGiver("student4InCourse1@gmail.tmt")
+                .withCommentGiverType(FeedbackParticipantType.OWN_TEAM_MEMBERS)
+                .withVisibilityFollowingFeedbackQuestion(true)
+                .withCourseId("idOfTypicalCourse1")
+                .withFeedbackSessionName("First feedback session")
+                .build();
+        relatedQuestion.setShowResponsesTo(Arrays.asList(FeedbackParticipantType.INSTRUCTORS));
+        response = dataBundle.feedbackResponses.get("response1ForQ1S1C1");
+        student = dataBundle.students.get("student1InCourse1");
+        studentsInTeam = roster.getTeamToMembersTable().getOrDefault(student.getTeam(), Collections.emptyList());
+        studentsEmailInTeam = studentsInTeam.stream()
+                .map(StudentAttributes::getEmail)
+                .collect(Collectors.toSet());
+
+        ______TS("success: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to receiver by default");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student1InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion,
+                commentFromS4C1ToR2Q2S1C1VisibilityFollowsQuestion));
+
+        ______TS("success: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to giver by default");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student4InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion,
+                commentFromS4C1ToR2Q2S1C1VisibilityFollowsQuestion));
+
+        ______TS("success: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to instructor");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("instructor1@course1.tmt", UserRole.INSTRUCTOR,
+                student, studentsEmailInTeam, response, relatedQuestion,
+                commentFromS4C1ToR2Q2S1C1VisibilityFollowsQuestion));
+
+        ______TS("failure: question is visible to instructor; visibility follows feedback question;"
+                + "comment is visible to teammate");
+        assertFalse(frcLogic.isResponseCommentVisibleForUser("instructor3@course1.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion,
+                commentFromS4C1ToR2Q2S1C1VisibilityFollowsQuestion));
+
+        relatedQuestion.setShowResponsesTo(Arrays.asList(FeedbackParticipantType.STUDENTS));
+
+        ______TS("success: question is visible to instructor; visibility follows feedback question;"
```

Comment:
is question visible to instructor from this point onwards?

Reply:
No, it is not.

Reply:
Then why is it written `question is visible to instructor` in the text?

Reply:
Sorry it was a careless mistake.